docs/articles/documentation/test-api/authentication/user-roles.md

```diff
@@ -10,6 +10,14 @@ to isolate authentication test actions and apply them easily whenever you need t
 
 A piece of logic that logs in a particular user is called a *role*. Define a role for each user account participating in your test.
 
+* [Why Use Roles](#why-roles)
+* [Create and Apply Roles](#create-and-apply-roles)
+* [Anonymous Role](#anonymous-role)
+* [Role Options](#role-options)
+  * [options.preserveUrl](#optionspreserveurl)
+* [Troubleshooting](#troubleshooting)
+  * [Test Actions Fail After Authentication](#test-actions-fail-after-authentication)
+
 ## Why Use Roles
 
 Roles are more than just another way to [extract reusable test logic](../../recipes/extract-reusable-test-code/README.md). They were specially designed for login operations and provide the following dedicated features:
```

Comment:
In addition to being used to [extract reusable test logic](../../recipes/extract-reusable-test-code/README.md), Roles were specially designed for login operations and provide the following dedicated features:

Reply:
I was trying to say that roles are not intended to be used to extract any random test pieces.
Though they may look like plain reusable functions, they were designed specially for authentication/login operations.

I just didn't want to say "don't" because we focus on benefits here.

So, maybe:

*Unlike other methods used to extract reusable test logic, roles were specially designed for login operations and provide the following dedicated features:*